File: 1-9/4/Main.kt
fun isPalindrome(number: Int): Boolean {
    val original = number
    var reversed = 0
    var temp = number
    while (temp > 0) {
        reversed = reversed * 10 + temp % 10
        temp /= 10
    }
    return original == reversed
}

fun main() {
    var currentMax = 0

    for (i in 100 until 1000) {
        for (j in 100 until 1000) {
            val product = i * j
            if (product > currentMax && isPalindrome(product)) {
                currentMax = product
            }
        }
    }

    println(currentMax)
}

File: 1-9/4/main.cpp
#include <iostream>

bool is_palindrome(int number) {
    int original = number;
    int reversed = 0;
    while (number > 0) {
        reversed = reversed * 10 + number % 10;
        number /= 10;
    }
    return original == reversed;
}

int main() {
    int current_max = 0;

    for (int i = 100; i < 1000; ++i) {
        for (int j = 100; j < 1000; ++j) {
            int product = i * j;
            if (product > current_max && is_palindrome(product))
                current_max = product;
        }
    }

    std::cout << current_max << std::endl;

    return 0;
}
